### PR TITLE
web: Convert card customisation-fetching to useResource

### PR DESCRIPTION
**Pull request by backspace**
I removed #1995 from #1987 so it would be focused, this brings it back.

### CODE REVIEW

**Comment by backspace** [comment]
looking into test failures 🧐

**Review by Aierie — APPROVED** [review]
Looks good!

File: packages/web-client/app/resources/card-customization.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+export class CardCustomization extends Resource<Args> {
+  // our output values
+  @tracked issuerName: string | undefined;
+  @tracked background: string | undefined;
+  @tracked patternColor: string | undefined;
+  @tracked textColor: string | undefined;
+  @tracked patternUrl: string | undefined;
+
+  // our derived state so you can check when the output values are ready, if you
```
Comment (Aierie) [inline packages/web-client/app/resources/card-customization.ts:25]: Unsure about use of the word, is this derived state? It isn't calculated based on other state in here.

Comment (backspace) [reply packages/web-client/app/resources/card-customization.ts:25]: I think it’s a reference to [similar properties](http://ember-concurrency.com/docs/derived-state) on Ember Concurrency tasks. But now that I think about it I think these comments were more to educate me than to be committed, so I’ll remove them 😆

**Pull request merged**